Dispatcher/UI/Homepage/HomepageViewModel.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+class HomepageViewModel {
+    
+    let repository = HomepageRepository()
+    var newsArray: [Article] = []
+    
+    private var currentPaginationPage = 1
+    private var totalPaginationPages = 1
+    
+    func fetchNewsFromAPI(completionHandler: @escaping () -> ()) {
+        
+        repository.fetchNewsFromAPI(currentPage: currentPaginationPage) { result, statusMsg in
+            
+            if statusMsg == nil {
+                self.currentPaginationPage += 1
+                self.totalPaginationPages = result!.totalPages
+                self.newsArray.append(contentsOf: result!.articles)
+            } else {
+                print(statusMsg ?? "error fetching articles")
```

Comment:
statusMsg can't be nil since you check it at the if.
Also pass it to UI, maybe you want to present the message.

Reply:
Fixed (also in other VMs)
@abenartz